funnel/views/project.py

```diff
@@ -123,12 +122,16 @@ class ProjectView(ProjectViewMixin, DraftViewMixin, UrlForView, ModelView):
     @render_with('project.html.jinja2')
     @requires_permission('view')
     def view(self):
-        rsvp_form = RsvpForm(obj=self.obj.rsvp_for(g.user))
+        current_rsvp = self.obj.rsvp_for(current_auth.user)
         transition_form = ProjectTransitionForm(obj=self.obj)
         schedule_transition_form = ProjectScheduleTransitionForm(obj=self.obj)
         project_save_form = SavedProjectForm()
+        rsvp_form = RsvpTransitionForm(
+            obj=self.obj.rsvp_for(current_auth.user, create=True, session_add=False)
```

Comment:
Don't use transitions to get the initial set of choices. Specify them elsewhere. See if you can extract them from a classmethod instead of an instance.